packages/dds/merge-tree/src/test/client.rollback.spec.ts

```diff
@@ -399,4 +399,96 @@ describe("client.rollback", () => {
 
 		assert.equal(segment.parent, undefined);
 	});
+	// similar to an existing test - is it worth checking that we can rollback to empty?
+	it("Should rollback multiple overlapping edits", () => {
+		client.insertTextLocal(0, "abcdefg");
+		client.insertTextLocal(3, "123");
+		client.annotateRangeLocal(2, 5, { foo: "bar" }, undefined);
+		client.removeRangeLocal(3, 7);
+
+		client.rollback?.({ type: MergeTreeDeltaType.REMOVE }, client.peekPendingSegmentGroups());
+		assert.equal(client.getText(), "abc123defg");
+		for (let i = 0; i < client.getText().length; i++) {
+			const props = client.getPropertiesAtPosition(i);
+			if (i >= 2 && i < 5) {
+				assert.equal(props?.foo, "bar");
+			} else {
+				assert(props === undefined || props.foo === undefined);
+			}
+		}
+
+		client.rollback?.({ type: MergeTreeDeltaType.ANNOTATE }, client.peekPendingSegmentGroups());
+		for (let i = 0; i < client.getText().length; i++) {
+			const props = client.getPropertiesAtPosition(i);
+			assert(props === undefined || props.foo === undefined);
+		}
+
+		client.rollback?.({ type: MergeTreeDeltaType.INSERT }, client.peekPendingSegmentGroups());
+		assert.equal(client.getText(), "abcdefg");
+		client.rollback?.({ type: MergeTreeDeltaType.INSERT }, client.peekPendingSegmentGroups());
+		assert.equal(client.getText(), "");
+	});
+	it("Should rollback multiple removes across split segments", () => {
+		client.insertTextLocal(0, "abcde");
+		client.insertTextLocal(3, "123");
+		client.insertTextLocal(4, "xyz");
+		client.removeRangeLocal(2, 5);
+		client.removeRangeLocal(3, 7);
```

Comment:
It would also be interesting to add another remove that surrounds a previously removed range. E.g. removeRangeLocal(2,4)